### DISCUSSION TITLE
Game Optimization

### DISCUSSION THREAD

**Discussion by UkrainianBanderasCat** [UkrainianBanderasCat/Digital-Logic-Sim2#27] [General]
We have a very bad optimization... We need to find where is the problem and fix it
![Screenshot from 2022-06-10 00-52-14](https://user-images.githubusercontent.com/48175036/173206427-31b30331-2def-46f0-891c-ba97f0fff1f2.png)
![Screenshot from 2022-06-10 00-51-55](https://user-images.githubusercontent.com/48175036/173206430-a576a17d-21fe-4d01-a7e5-2afbb0197dd1.png)

**Comment by firecerne** [comment]
I don't know the problem , but from what I understood reading the code, each chip inside a chip inside a chip inside your chip builder is calculated independently. So, for more complex chips, the number adds up quickly. The behaviour of each of these chips is calculated, using the 2 basic functions.
If we changed drastically the system, it could be solved. I thought about calculating the truth table for each chip and storing that instead, but I think it would be very hard for more complex chips.

**Reply by AlphaCodingPilot** [reply]
The problem is with chips storing data like memory. You whould also need to store the internal state of each chip.

**Comment by hodytone** [comment]
If we find solution for this problem - it would be The Biggest Invention of The World Engeneering!

**Comment by hodytone** [comment]
I think we can make a little improvement by separating the chip editing/visualization and signal processing mechaniс.

**Reply by UkrainianBanderasCat** [reply]
Ok
